### PR TITLE
Fixed a bug in joint intent slot infrence files.

### PR DESCRIPTION
**Pull request by VahidooX**
Signed-off-by: VahidooX <vahid.noroozi@gmail.com>

### CODE REVIEW

File: examples/nlp/joint_intent_slot_with_bert.py [hunk 1]
```diff
@@ -53,6 +53,7 @@
 parser.add_argument("--intent_loss_weight", default=0.6, type=float)
 parser.add_argument("--class_balancing", default="regular", type=str,
                     choices=["regular", "weighted_loss"])
+parser.add_argument("--eval_freq", default=1, type=int)
```
Comment (chiphuyen) [inline examples/nlp/joint_intent_slot_with_bert.py]: looks like it's `eval_freq_by_epoch`?

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: Actually evaluation is making multi gpu training on ngc super slow (more than 3 times slower). Our guess is that evaluation process is the main cause. So we want evaluation to be done every couple of epochs. How can we have such a thing?

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: We can have eval_step_freq instead of eval_epoch_freq but it would be hard to calculate the number of steps for each epoch for each dataset.

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: There is another way too. We can make it a float number so that for instance by setting 0.2 for, we get evlauation 5 times an epoch.

Comment (chiphuyen) [reply examples/nlp/joint_intent_slot_with_bert.py]: what do you think @okuchaiev

Comment (tkornuta-nvidia) [reply examples/nlp/joint_intent_slot_with_bert.py]: Actually doing few evaluation per epoch appears to be quite useful. The trick is that you do not use the whole validation set, but its subset (Andrew Ng is call that the "Eyeball dev set"). We relied on that so extensively that I have implemented a second type of trainer in PTP that implemented that "training methodology"...

Comment (okuchaiev) [reply examples/nlp/joint_intent_slot_with_bert.py]: I'm in favour of "eval_step_freq" parameter. Float steps seem weird

Comment (chiphuyen) [reply examples/nlp/joint_intent_slot_with_bert.py]: i'm ok with eval_step_freq. i'm just concerned about this line:

`eval_step=steps_per_epoch*args.eval_freq`

anyway, i don't think it's a blocking matter. we can merge this pr and fix it later too.

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: This "Eyeball dev set" is helpful especially for very large datasets. For now, datasets available for intent detection/slot tagging are not that large. But good to have that option for future.

eval_step_freq is more flexible, but how can we have set to have evaluation each epoch. It needs recalculation each time that batch_size or dataset is changed which happens a lot with this task. 

How about having both options eval_step_freq and eval_epoch_freq and prioritize one if both are set?

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: I took a look at the callback functions for evaluation. It seems you have the same idea but it is not implemented completely yet. There is a TODO in EvaluatorCallback class for this and eval_epoch is not working yet.

Comment (VahidooX) [reply examples/nlp/joint_intent_slot_with_bert.py]: Kept eval_step_freq for now.

File: examples/nlp/joint_intent_slot_infer_b1.py [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 import numpy as np
 from pytorch_transformers import BertTokenizer
```
Comment (chiphuyen) [inline examples/nlp/joint_intent_slot_infer_b1.py]: i think we've decided to move to transformers?

Comment (ekmb) [reply examples/nlp/joint_intent_slot_infer_b1.py]: I think we decided to wait with transformers since it breaks Bert pretraining

Comment (blisc) [reply examples/nlp/joint_intent_slot_infer_b1.py]: I am assuming this is resolved
